File: backend/src/platform/schemas/invitation.py
from datetime import UTC, datetime

from pydantic import BaseModel, ConfigDict, computed_field


class InvitationInviterOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    email: str


class InvitationRoleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str


class InvitationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    email: str
    role_ids: list[int]
    roles: list[InvitationRoleOut] = []
    invited_by: InvitationInviterOut | None
    created_at: datetime
    expires_at: datetime

    @computed_field
    @property
    def is_expired(self) -> bool:
        return self.expires_at <= datetime.now(UTC)

File: frontend/src/platform/types/invitation.ts
export interface InvitationRoleOut {
  id: number
  name: string
}

export interface InvitationInviterOut {
  id: number
  name: string
  email: string
}

export interface InvitationOut {
  id: number
  email: string
  role_ids: number[]
  /** role_ids resolved to the org's current roles (deleted roles omitted). */
  roles: InvitationRoleOut[]
  /** null when the inviting user has since been deleted. */
  invited_by: InvitationInviterOut | null
  created_at: string
  expires_at: string
  is_expired: boolean
}
